src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1771,6 +1789,46 @@ private void RegisterHomeWorkspace(HomeWorkspaceModel newWorkspace)
             };
         }
 
+        /// <summary>
+        /// Collects assemblies used by the current workspace
+        /// </summary>
+        /// <returns></returns>
+        private HashSet<AssemblyName> OnCollectingAssembliesUsed()
+        {
+            var assemblyNames = new HashSet<AssemblyName>();
+            foreach(var node in CurrentWorkspace.Nodes)
+            {
+                // Get zerotouch assemblies
+                if (node.GetType() == typeof(DSFunction))
+                {
+                    // Get FunctionDescriptors for this zerotouch node
+                    var descriptors = LibraryServices.GetAllFunctionDescriptors(node.CreationName.Split('@')[0]);
+                    if (descriptors != null)
+                    {
+                        foreach (FunctionDescriptor fd in descriptors)
```

Comment:
why even do this comparison if the descriptor is not a package member? Why not skip all this?

Reply:
Good point. Moved the `IsPackageMember` check up